app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -10,12 +10,25 @@ import android.util.AttributeSet
 import android.view.View
 import androidx.appcompat.app.AppCompatActivity
 import mozilla.components.concept.engine.EngineView
+import mozilla.components.feature.intent.IntentProcessor
+import mozilla.components.support.utils.SafeIntent
+import org.mozilla.fenix.browser.BrowserFragment
 import org.mozilla.fenix.ext.components
 
-class HomeActivity : AppCompatActivity() {
+open class HomeActivity : AppCompatActivity() {
+    private var sessionId: String? = null
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_home)
+
+        if (savedInstanceState == null) {
+            sessionId = SafeIntent(intent).getStringExtra(IntentProcessor.ACTIVE_SESSION_ID)
+            supportFragmentManager?.beginTransaction()?.apply {
+                replace(R.id.container, BrowserFragment.create(sessionId))
+                commit()
+            }
```

Comment:
Can this transaction be expressed in the navigation library? Or does it make sense to have a completely separate activity here and bite the bullet of possibly some duplication (that could be cleaned up at a later stage)?

Reply:
I'll take a look at this today and play around with the navigation part of it